File: iot-core/test-devices/device-send-telemetry.js
const mqtt = require("mqtt")

let gps = require('gps-simulator/gps-simulator.js');
let gpsData = require('gps-simulator/gps-simulator-data.js');

// There are 4 routes in gpsData, named routes01 -> routes04
// Each route, there are 2 directions: AB and BA
// So to get one route data, just use: gpsData.routes01.AB, for example.

let DEVICE_A_ACCESS_TOKEN = "0520f900a18e12a4d12c"
let DEVICE_B_ACCESS_TOKEN = "10cc9d574a42aa2799c0"

let deviceAGpsSimulator = new gps.GpsSimulator(gpsData.routes[Math.floor(Math.random() * gpsData.routes.length)].AB);
let deviceBGpsSimulator = new gps.GpsSimulator(gpsData.routes[Math.floor(Math.random() * gpsData.routes.length)].AB);

const client = mqtt.connect({host: "fiot.me", port: 1883})
client.on("connect", function () {
  console.log("MQTT connected.")

  deviceAGpsSimulator.start(function (position, beStopped, movableObject, currentRouteIndex) {
    const randomTemp = Math.floor(Math.random() * (80 - (-20) + 1) + (-20))
    let str = "Route " + currentRouteIndex + ", speed " + movableObject.velocity * 3.6 + " km/h, temperature " + randomTemp;
    console.log('[ ' + new Date() + ' ] ' + str);

    // Do something you want with gps_sensor
    client.publish('/v1/devices/telemetry/request', JSON.stringify({
      "token": DEVICE_A_ACCESS_TOKEN,
      "json": {
        "temperature": randomTemp,
        "latitude": position.latitude,
        "longitude": position.longitude,
        "speed": movableObject.velocity * 3.6,
      }
    }))

  });

  // deviceBGpsSimulator.start(function (position, beStopped, movableObject, currentRouteIndex) {
  //   const randomTemp = Math.floor(Math.random() * (80 - (-20) + 1) + (-20))
  //   let str = "Route " + currentRouteIndex + ", speed " + movableObject.velocity * 3.6 + " km/h, temperature " + randomTemp;
  //   console.log('[ ' + new Date() + ' ] ' + str);
  //
  //   // Do something you want with gps_sensor
  //   client.publish('/v1/devices/telemetry/request', JSON.stringify({
  //     "token": DEVICE_B_ACCESS_TOKEN,
  //     "json": {
  //       "temperature": randomTemp,
  //       "latitude": position.latitude,
  //       "longitude": position.longitude
  //     }
  //   }))
  //
  // });
})

File: iot-core/test-devices/device-send-telemetry.py
import paho.mqtt.client as mqtt
from time import sleep
import random

import json
# import geocoder
# import pyttsx3
#
# engine = pyttsx3.init()
# engine.say("Whetever you want the program to say")
# engine.runAndWait()

TOPIC_PUB = '/v1/devices/telemetry/request'
# TOPIC_SUB = '/v1/device/telemetry/response'
HOST = 'localhost'
PORT = 1883

def on_connect(client, userdata, flags, rc):
    print("rc_code:", rc)
    # client.subscribe(TOPIC_SUB)

def on_message(client, userdata, message):
    print("Topic: " + message.topic + "\nMessage: " + str(message.payload.decode('utf-8')))

client = mqtt.Client()

client.on_connect = on_connect
# client.on_message = on_message

client.connect(HOST, PORT)
client.loop_start()

while True:
    x = random.randrange(-20, 80)
    latitude = g.latlng[0]
    longitude = g.latlng[1]

    msg = {
        "token": "b105b10f1ad550b81fb0",
        "json": {
            "temperature": x,
        },
    }
    print(msg)
    
    client.publish(TOPIC_PUB, json.dumps(msg))

    sleep(1.5)
